XMonad/Actions/EasyMotion.hs

```diff
@@ -0,0 +1,378 @@
+         , borderPx    :: Int                                -- ^ Width of border in pixels
+         , maxChordLen :: Int                                -- ^ Maximum chord length. Use 0 for no maximum.
+         }
+
+instance Default EasyMotionConfig where
+  def =
+    EMConf { txtCol      = "#ffffff"
+           , bgCol       = "#000000"
+           , overlayF    = proportional (0.3::Double)
+           , borderCol   = "#ffffff"
+           , sKeys       = AnyKeys [xK_s, xK_d, xK_f, xK_j, xK_k, xK_l]
+           , cancelKey   = xK_q
+           , borderPx    = 1
+           , maxChordLen = 0
+#ifdef XFT
+           , font        = "xft:Sans-100"
+#else
+           , font        = "-misc-fixed-*-*-*-*-200-*-*-*-*-*-*-*"
+#endif
+           }
+
+-- | Create overlay windows of the same size as the window they select
+fullSize :: Position -> Rectangle -> Rectangle
+fullSize _ = id
+
+-- | Create overlay windows a proportion of the size of the window they select
+proportional :: RealFrac f => f -> Position -> Rectangle -> Rectangle
+proportional f th r = Rectangle { rect_width  = newW
+                                , rect_height = newH
+                                , rect_x      = rect_x r + fi (rect_width r - newW) `div` 2
+                                , rect_y      = rect_y r + fi (rect_height r - newH) `div` 2 }
+  where
+    newH = max (fi th) (round $ f * fi (rect_height r))
+    newW = newH
+
+-- | Create fixed-size overlay windows
+fixedSize :: (Integral a, Integral b) => a -> b -> Position -> Rectangle -> Rectangle
+fixedSize w h th r = Rectangle { rect_width  = rw
+                               , rect_height = rh
+                               , rect_x      = rect_x r + fi (rect_width r - rw) `div` 2
+                               , rect_y      = rect_y r + fi (rect_height r - rh) `div` 2 }
+  where
+    rw = max (fi w) (fi th)
+    rh = max (fi h) (fi th)
+
+-- | Create overlay windows the minimum size to contain their key chord
+textSize :: Position -> Rectangle -> Rectangle
+textSize th r = Rectangle { rect_width  = fi th
+                          , rect_height = fi th
+                          , rect_x      = rect_x r + (fi (rect_width r) - fi th) `div` 2
+                          , rect_y      = rect_y r + (fi (rect_height r) - fi th) `div` 2 }
+
+-- | Create overlay windows the full width of the window they select, the minimum height to contain
+--   their chord, and a proportion of the distance from the top of the window they select
+bar :: RealFrac f => f -> Position -> Rectangle -> Rectangle
+bar f th r = Rectangle { rect_width  = rect_width r
+                       , rect_height = fi th
+                       , rect_x      = rect_x r
+                       , rect_y      = rect_y r + round (f' * (fi (rect_height r) - fi th)) }
+                         -- clamp f in [0,1] as other values will appear to lock up xmonad
+                         -- as the overlay will be displayed off-screen
```

Comment:
I feel like this may be nicer as documentation for the definition of `f'`

Reply:
I'm not sure if you meant dedented, I suspect this comment has suffered from autoindent (and my subsequent negligence). I've dedented it, let me know if you meant something else.

Reply:
You're completely right; thanks!